### PR TITLE
Add CI

### PR DESCRIPTION
**Pull request by swanderz**
To Do:
- [ ] Add Git to actual docker image
- [x] figure out why submodules aren't auto-installing? @jtcohen6 halp plz
- [ ] add a local version of SQL server for testing as well

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,18 +3,29 @@ version: 2.1
...
           command: |
             git submodule init
             git submodule update --remote
-      
+
+      - run:
+          name: "TEMP: install dbt-utils"
+          command: |
+            git clone --recursive https://github.com/fishtown-analytics/dbt-utils 
+
```
Comment (swanderz) [inline .circleci/config.yml:28]: I don't think the `git clone` step I added is supposed to be necessary, but I had to run it to get it to work.

Comment (jtcohen6) [reply .circleci/config.yml:28]: I think I did this wrong when I first set it up. Try running this locally:
```
git submodule add https://github.com/fishtown-analytics/dbt-utils
```
and then commit + push that change to this branch

**Comment by jtcohen6** [comment]
Looks like the CI failures have to do with the `dbt seed` step. These are tricky:
- Cannot find data type 'timestamp'.
- Cannot insert an explicit value into a timestamp column. Use INSERT with a column list to exclude the timestamp column, or insert a DEFAULT into the timestamp column.
- A table can only have one timestamp column. Because table 'data_dateadd' already has one, the column 'result' cannot be added.

It's possible that we'll need to:
- Disable the seed coming from `dbt_utils_integration_tests`
- Recreate the seed (different name) + staging view (same name) to replicate the test setup

**Comment by swanderz** [comment]
> * Cannot find data type 'timestamp'.
> 
> * Cannot insert an explicit value into a timestamp column. Use INSERT with a column list to exclude the timestamp column, or insert a DEFAULT into the timestamp column.
> 
> * A table can only have one timestamp column. Because table 'data_dateadd' already has one, the column 'result' cannot be added.

yeah these definitely seem T-SQL specific. I'll ask @NandanHegde15 to take a look at those errors. I agree we might have to make different versions of the csvs if we can't fix it adapter-side.

meanwhile, is there a way to disable that set of seeds so we can get some feedback on which macros work and which don't?

**Comment by jtcohen6** [comment]
> meanwhile, is there a way to disable that set of seeds so we can get some feedback on which macros work and which don't?

I think you can update what's [here](https://github.com/dbt-msft/tsql-utils/blob/7ede42a031f3e8787272f306eb026d282c9c7cc7/integration_tests/dbt_utils/dbt_project.yml#L22-L35) to be:
```yml
seeds:
  dbt_utils_integration_tests:
    cross_db:
      data_date_trunc:
        +enabled: false
      data_dateadd:
        +enabled: false
      data_datediff:
        +enabled: false
      data_last_day:
        +enabled: false
    etc:
      data_people:
        +enabled: false
    sql:
      data_get_query_results_as_dict
        +enabled: false
      data_safe_add
        +enabled: false
    schema_tests:
      data_test_mutually_exclusive_ranges_with_gaps:
        +enabled: false
      data_test_not_null_where:
        +enabled: false

models:
  dbt_utils_integration_tests:
    cross_db_utils:
      +enabled: true
      test_date_trunc:
        +enabled: false
      test_dateadd:
        +enabled: false
      test_datediff:
        +enabled: false
      test_last_day:
        +enabled: false
    datetime:
      +enabled: true
    materializations:
      +enabled: false
    schema_tests:
      +enabled: false
    sql:
      +enabled: true
      test_safe_add:
        +enabled: false
    web:
      +enabled: false
```

**Comment by swanderz** [comment]
The issue I'm seeing now is that in T-SQL, the type `timestamp` is really `datetime` (`timestamp` is a special thing in T-SQL that's mostly deprecated now). So if I want to change the column types in [dbt-utils's 'integration_tests/dbt_project.yml`](
https://github.com/fishtown-analytics/dbt-utils/blob/ebda5845df85194da981fa2c9da499750da51bd3/integration_tests/dbt_project.yml#L30-L42) how might one do that?

```YAML
seeds:

  +quote_columns: false
  dbt_utils_integration_tests:

    cross_db:
      data_date_trunc:
        +column_types:
          updated_at: timestamp
          day: date
          month: date

      data_dateadd:
        +column_types:
          from_time: timestamp
          result: timestamp

      data_datediff:
        +column_types:
          first_date: timestamp
          second_date: timestamp
```

**Comment by jtcohen6** [comment]
the way you just did it is what I was going to recommend :)

File: integration_tests/ci/sample.profiles.yml [hunk 2]
```diff
@@ -22,3 +22,15 @@ integration_tests:
...
       threads: 1
+    azuresql:
+      type: sqlserver
+      driver: "ODBC Driver 17 for SQL Server"
+      port: 1433
+      host: "{{ env_var('DBT_AZURESQL_SERVER') }}"
+      database: "{{ env_var('DBT_AZURESQL_DB') }}"
+      username: "{{ env_var('DBT_AZURESQL_UID') }}"
+      password: "{{ env_var('DBT_AZURESQL_PWD') }}"
+      schema: dbt_external_tables_integration_tests_azuresql
```
Comment (jtcohen6) [inline integration_tests/ci/sample.profiles.yml:33]:
```suggestion
      schema: tsql_utils_integration_tests_azuresql
```

**Comment by swanderz** [comment]
ok now we're getting juicy, actionable information! thanks @jtcohen6!

```log
Database Error in model test_current_timestamp (models/cross_db_utils/test_current_timestamp.sql)
  ('42000', "[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Incorrect syntax near '::'. (102) (SQLExecDirectW)")
  compiled SQL at target/run/dbt_utils_integration_tests/models/cross_db_utils/test_current_timestamp.sql

Database Error in model test_current_timestamp_in_utc (models/cross_db_utils/test_current_timestamp_in_utc.sql)
  ('42000', "[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Incorrect syntax near '::'. (102) (SQLExecDirectW)")
  compiled SQL at target/run/dbt_utils_integration_tests/models/cross_db_utils/test_current_timestamp_in_utc.sql

Database Error in model test_date_spine (models/datetime/test_date_spine.sql)
  ('42000', "[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Incorrect syntax near the keyword 'with'. (156) (SQLExecDirectW)")
  compiled SQL at target/run/dbt_utils_integration_tests/models/datetime/test_date_spine.sql

Database Error in model test_date_trunc (models/cross_db_utils/test_date_trunc.sql)
  ('42000', "[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]'date_trunc' is not a recognized built-in function name. (195) (SQLExecDirectW)")
  compiled SQL at target/run/dbt_utils_integration_tests/models/cross_db_utils/test_date_trunc.sql

Database Error in model test_generate_series (models/sql/test_generate_series.sql)
  ('42000', "[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Incorrect syntax near the keyword 'with'. (156) (SQLExecDirectW)")
  compiled SQL at target/run/dbt_utils_integration_tests/models/sql/test_generate_series.sql

Database Error in model test_get_column_values (models/sql/test_get_column_values.sql)
  ('42000', '[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Each GROUP BY expression must contain at least one column that is not an outer reference. (164) (SQLExecDirectW)')

Database Error in model test_get_relations_by_pattern (models/sql/test_get_relations_by_pattern.sql)
  ('42000', "[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Incorrect syntax near '-'. (102) (SQLExecDirectW)")

Database Error in model test_get_relations_by_prefix_and_union (models/sql/test_get_relations_by_prefix_and_union.sql)
  ('42000', "[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Incorrect syntax near '-'. (102) (SQLExecDirectW)")

Database Error in model test_groupby (models/sql/test_groupby.sql)
  ('42000', '[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Each GROUP BY expression must contain at least one column that is not an outer reference. (164) (SQLExecDirectW)')
  compiled SQL at target/run/dbt_utils_integration_tests/models/sql/test_groupby.sql

Database Error in model test_hash (models/cross_db_utils/test_hash.sql)
  ('42S22', "[42S22] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Invalid column name 'field'. (207) (SQLExecDirectW)")
  compiled SQL at target/run/dbt_utils_integration_tests/models/cross_db_utils/test_hash.sql

Database Error in model test_last_day (models/cross_db_utils/test_last_day.sql)
  ('42000', "[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]'date_trunc' is not a recognized built-in function name. (195) (SQLExecDirectW)")
  compiled SQL at target/run/dbt_utils_integration_tests/models/cross_db_utils/test_last_day.sql

Database Error in model test_length (models/cross_db_utils/test_length.sql)
  ('42000', "[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]'length' is not a recognized built-in function name. (195) (SQLExecDirectW)")
  compiled SQL at target/run/dbt_utils_integration_tests/models/cross_db_utils/test_length.sql

Database Error in model test_position (models/cross_db_utils/test_position.sql)
  ('42000', "[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Incorrect syntax near 'string_text'. (102) (SQLExecDirectW)")
  compiled SQL at target/run/dbt_utils_integration_tests/models/cross_db_utils/test_position.sql

Database Error in model test_split_part (models/cross_db_utils/test_split_part.sql)
  ('42000', "[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]'split_part' is not a recognized built-in function name. (195) (SQLExecDirectW)")
  compiled SQL at target/run/dbt_utils_integration_tests/models/cross_db_utils/test_split_part.sql

Database Error in model test_surrogate_key (models/sql/test_surrogate_key.sql)
  ('42S22', "[42S22] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Invalid column name 'field'. (207) (SQLExecDirectW)")
  compiled SQL at target/run/dbt_utils_integration_tests/models/sql/test_surrogate_key.sql

Database Error in model test_width_bucket (models/cross_db_utils/test_width_bucket.sql)
  ('42000', "[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]'mod' is not a recognized built-in function name. (195) (SQLExecDirectW)")
  compiled SQL at target/run/dbt_utils_integration_tests/models/cross_db_utils/test_width_bucket.sql

Done. PASS=14 WARN=0 ERROR=16 SKIP=0 TOTAL=30

```

**Pull request merged**